bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/handler/DoorbellHandler.java

```diff
@@ -411,6 +429,23 @@ private void stopMotionOffJob() {
         }
     }
 
+    @SuppressWarnings({ "unchecked", "null" })
+    private void startAudioSink() {
+        final DoorbellHandler thisHandler = this;
+        // Register an audio sink in openhab
+        logger.trace("Registering an audio sink for this {}", thing.getUID());
+        audioSinkRegistration = (ServiceRegistration<AudioSink>) bundleContext
+                .registerService(AudioSink.class.getName(), new DoorbirdAudioSink(thisHandler), new Hashtable<>());
+    }
+
+    private void stopAudioSink() {
+        // Unregister the doorbird audio sink
+        if (audioSinkRegistration != null) {
+            logger.trace("Unregistering the audio sync service for the doorbird thing {}", getThing().getUID());
+            audioSinkRegistration.unregister();
+        }
+    }
```

Comment:
Generally, it is rather done in the thing handler factory.
That being said, it is probably correct to do it like that directly in the thing handler.

Reply:
It seems less convenient ?
In the factory, managing the life cycle of the audiosink seems complicated, because the audiosink life is completely linked with the handler activation/deactivation. The HandlerFactory createHandler / unregisterHandler has not the same meaning ?

Reply:
In fact the audio sink life cycle is linked to the thing handler life cycle and it remains true with your implementation choice as you start it in thing handler handler initialize method and stop it in thing handler dispose method.
You made the choice to do it directly in the thing handler while until now all implementations I saw made the choice to do it in the thing handler (when creating/removing the thing handler).
IMHO the result is the same, even if I have a small preference for the handler factory choice.